### PR TITLE
feat: Image drag and drop for ImageInput

### PR DESCRIPTION
**Pull request by Abhi-Bohora**
## Changes
Linked Issue: (https://github.com/dailydotdev/daily/issues/1369)

### Describe what this PR does
As said in the issue `We have an ImageInput for uploading images (as the name implies), but it lacks drag-and-drop functionality`. so this means the user cannot simply drag and drop their images so this PR aims to fix that

Any suggestions, Improvements 🙏

https://github.com/user-attachments/assets/583804c7-6c3b-447c-bcef-6589940b996e

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### CODE REVIEW

File: packages/shared/src/components/fields/ImageInput.tsx [hunk 1]
```diff
@@ -102,6 +103,32 @@ function ImageInput({
     onChange?.(base64, file);
   };
 
+  const onDragOver = (event: DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const onDrop = async (event: DragEvent<HTMLButtonElement>) => {
```
Comment (nensidosari) [inline packages/shared/src/components/fields/ImageInput.tsx:110]: Seems like both onFileChange and onDrop do almost the same things after they have the file. Would be nice to unify these functions rather then repeat code.

**Comment by nensidosari** [comment]
Hi @Abhi-Bohora thank you for your contribution. Have left just one small comment regarding code, other then that things seem good :D

**Review by sshanzel — APPROVED** [review]
I don't have anything to add on top of Nensi's comment so I'm approving. Really looks amazing with the screen recording 😍

File: packages/shared/src/components/fields/ImageInput.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/ImageInput.tsx:110]: Maybe introduce a function named `validateFileInput` or something to contain the logic.

**Pull request merged**